modules/mergesort/test/test.cpp

```diff
@@ -0,0 +1,13 @@
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE sort_tests
+#include <boost/test/unit_test.hpp>
+#include "mergesort.h" 
+
+BOOST_AUTO_TEST_SUITE(merge_sort_test);
+BOOST_AUTO_TEST_CASE(merge_sort_test_case1) {
+    int test[] = {-5,2,6,4,12,-7};
+    int testans[] = {-7,-5,2,4,6,12}
+    merge_sort(test,sizeof(test)/sizeof(test[0]));
+    BOOST_CHECK(test == testans);
```

Comment:
Note that on C-style arrays operator== will compare addresses rather than values, so this check will always fail. 
To compare them you should either use a higher level implementation such as std::vector or std::array, or there's also an [std::equal](http://www.cplusplus.com/reference/algorithm/equal/) function in <algorithm> which can be used in conjunction with std::begin and std::end from <iterator> to compare arrays.